src/openms/include/OpenMS/DATASTRUCTURES/QTCluster.h

```diff
@@ -119,115 +118,97 @@ namespace OpenMS
-     * center point does not have any peptide id associated with it. In this
-     * case, it is not clear which peptide id the current cluster should use.
-     * The function thus iterates through all possible peptide ids and selects
-     * the one producing the best cluster.
-     *
-     * This function needs access to all possible neighbors for this cluster
-     * and thus can only be run when tmp_neighbors_ is filled (which is during
-     * the filling of a cluster). The function thus cannot be called after
-     * finalizing the cluster.
-     *
-     * @returns The total distance between cluster elements and the center.
+     * @brief Class to store the bulk internal data (neihgbors, annotations, etc.)
+     * 
+     * Has no functionality without a QTCluster (-head) pointing to it.
+     * Create object of this class before calling constructor of QTCluster
      */
-    double optimizeAnnotations_();
-
-public:
+    class OPENMS_DLLAPI Data_
+    {
+      /// QTCluster can access and manage all members of Data_
+      friend class QTCluster;
+
+      /// Pointer to the cluster center
+      GridFeature* center_point_;
+
+      /// unique id of this cluster
+      Size id_;
+
+      /**
+       * @brief Map that keeps track of the best current feature for each map
+       *
+       */
+      NeighborMap neighbors_;
+
+      /**
+       * @brief Temporary map tracking *all* neighbors
+       *
+       * For each input run, a multimap which contains pointers to all
+       * neighboring elements and the respective distance.
+       *
+       */
+      NeighborMapMulti tmp_neighbors_;
+
+      /// Maximum distance of a point that can still belong to the cluster
+      double max_distance_;
+
+      /// Number of input maps
+      Size num_maps_;
+
+      /// x coordinate in the grid cell
+      Int x_coord_;
+
+      /// y coordinate in the grid cell
+      Int y_coord_;
+
+      /**
+       * @brief Set of annotations of the cluster
+       *
+       * The set of peptide sequences that is compatible to the cluster center
+       * and results in the best cluster quality.
+       */
+      std::set<AASequence> annotations_;
+    };
 
     /**
      * @brief Detailed constructor
+     * @param data Pointer to internal data
      * @param center_point Pointer to the center point
      * @param num_maps Number of input maps
      * @param max_distance Maximum allowed distance of two points
      * @param use_IDs Use peptide annotations?
+     * @param id Unique ID of this cluster
      */
-    QTCluster(GridFeature* center_point, Size num_maps,
-              double max_distance, bool use_IDs, 
-              Int x_coord, Int y_coord);
 
-    /// Destructor
-    virtual ~QTCluster();
+    QTCluster(Data_* data, GridFeature* center_point, 
+              Size num_maps, double max_distance, 
+              bool use_IDs, Int x_coord, Int y_coord,
+              Size id);
+
+    /// Default constructor not accessible
```

Comment:
explain why not accessible

Reply:
This is also from before. Would it be okay to delete the comment and then one knows that the object should only be constructed with the custom constructor?

Reply:
Added explanation